README.md

```diff
@@ -9,13 +9,36 @@ pip install .
 ```
 in your preferred Python venv.
 
-You can then import the lego command and run any function that you can run from the CLI:
-```
+
+## Usage
+
+You can import the lego command and run any function that you can run from the CLI:
+
+```python
 from lepy import run_lego_command
 test_env = {"NAMECHEAP_API_USER": "user", "NAMECHEAP_API_KEY": "key"}
-run_lego_command("something@gmail.com", "127.0.0.1", "/path/to/csr.pem", "namecheap", test_env)
+run_lego_command("something@gmail.com", "https://localhost/directory", "-----BEGIN CERTIFICATE REQUEST----- ...", "namecheap", test_env)
 ```
 
+| Argument | Description |
+| -------- | ----------- |
+| `email`  | The provided email will be registered to LetsEncrypt. It may receive some emails notifying the user about certificate expiry. |
+| `server` | This is the full URL of a server that implements the ACME protocol. While letsencrypt is the most common one, there are other programs that provide this facility like Vault. |
+| `csr`    | This must be a PEM string in bytes that is user generated and valid as according to the ACME server that is being provided above. Many providers have different requirements for what is allowed to be in the fields of the CSR.|
+| `plugin` | The plugin is a string that's supported by LEGO. The full list is located [here](https://go-acme.github.io/lego/dns/). On top of the LEGO provided ones, we have an extra plugin called `http` that will allow users to use HTTP01 and TLSALPN01 challenges. |
+| `env`    | The env is a dictionary mapping of strings to strings that will be loaded into the environment for LEGO to use. All plugins require some configuration values loaded into the environment. You can find them [here](https://go-acme.github.io/lego/dns/) |
+
+
+On top of the environment variables that LEGO supports, we have some extra ones that we use to configure the library:
+
+| Key | Description |
+| --- | ----------- |
+| `SSL_CERT_FILE`   | Path to a CA certificate file for lepy to trust. This can be used for trusting the certificate of the ACME server provided. |
+| `HTTP01_URL`      | The interface to be used for the HTTP01 challenge if the plugin is chosen. Any interface by default. |
+| `HTTP01_PORT`     | The port to be used for the HTTP01 challenge if the plugin is chosen. 80 by default.|
+| `TLSALPN01_ULR`   | The interface to be used for the TLSALPN01 challenge if the plugin is chosen. Any interface by default. |
```

Comment:
Same comment, and there is also a typo in the name of the variable.

Reply:
updated